sesman/session.c

```diff
@@ -492,6 +494,27 @@ session_start(long data,
         return 0;
     }
 
+#ifdef HAVE_SYS_PRCTL_H
+    /*
+     * Check if we are running inside AppArmor confinement.
+     * If AppArmor is loaded, _and_ we are confined, then
+     * don't use PR_SET_NO_NEW_PRIVS later.
+     *
+     * (We do this check now rather than later to avoid
+     * opening the /proc file as a user different from
+     * the one owning it, so that the access can be
+     * allowed with an "owner"-qualified AppArmor rule.)
+     */
+    fd = g_file_open_ro("/proc/self/attr/apparmor/current");
+    if (fd != -1)
+    {
+        char buf[16] = {0};
+        g_file_read(fd, buf, sizeof(buf) - 1);
+        g_file_close(fd);
+        process_is_unconfined = !g_strcmp(buf, "unconfined\n");
+    }
+#endif
+
```

Comment:
I can see why you're doing this this way, but this is a platform-specific test, so I don't think it belongs in-line in this way.

In fact, the `prctl()` call shouldn't really be here either - I know that's not your fault.

How about adding a call `g_set_no_new_privs()` to common/os_calls.c? Than can be a no-op on anything other than linux, and all the code related to this area can go in that.

Did I explain that well enough?

Reply:
Hi @matt335672, I understand your concern. I think an abstracted NoNewPrivileges call is a fine idea.

I'll update this PR, and add in a commit with the `g_file_open_rw()` change as well. Allow me a day or so.